src/three-components/TextureUtils.ts

```diff
@@ -233,7 +232,10 @@ export default class TextureUtils extends EventDispatcher {
         }
       }
 
-      if (options.pmrem) {
+      // Apply PMREM to HDR textures, which if loaded from file have
+      // RGBEEncoding, and if generated are HalfFloatType
+      if (environmentMap.encoding === RGBEEncoding ||
+          environmentMap.type === HalfFloatType) {
```

Comment:
Woah, we weren't even applying PMREM to the generated map? That... doesn't seem right...